lib/puppet/provider/sensu_user/sensuctl.rb

```diff
@@ -52,32 +52,9 @@ def password_insync?(user, password)
     if password.is_a?(Array)
       password = password[0]
     end
-    # Use this approach once merged: https://github.com/sensu/sensu-go/pull/2278
-    #ret = execute(['sensuctl', 'user', 'test-creds', user, '--password', password], failonfail: false)
-    #exitstatus = ret.exitstatus
-    #exitstatus == 0
-    config = load_config(config_path)
-    api_url = config['api-url']
-    if api_url =~ /^https/
-      use_ssl = true
-    else
-      use_ssl = false
-    end
-    uri = URI(api_url)
-    http = Net::HTTP.new(uri.hostname, uri.port)
-    if Puppet[:debug]
-      http.set_debug_output($stdout)
-    end
-    http.use_ssl = use_ssl
-    if use_ssl
-      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
-    end
-    request = Net::HTTP::Get.new('/auth')
-    request.add_field("Accept", "application/json")
-    request.basic_auth(user, password)
-    response = http.request(request)
-    Puppet.debug("GET /auth returned #{response.code}")
-    response.kind_of?(Net::HTTPSuccess)
+    ret = execute(['sensuctl', 'user', 'test-creds', user, '--password', password], failonfail: false)
+    exitstatus = ret.exitstatus
```

Comment:
why get the status then set it to 0?

Reply:
do we have a test that shows what happens when you supply the wrong password and try to change passwords?

Reply:
It would have to be an acceptance test, unit testing this wouldn't really prove much since when changing passwords we just pass old and new with no validation.  We could, in theory, validate the old password first using same function.

Reply:
Added commit that tests the `old_password` before attempting to change the password.  Also added corresponding unit and acceptance tests.